### PR TITLE
core/manager: clean up taint strings

### PR DESCRIPTION
**Pull request by YHNdnzj**

### CODE REVIEW

File: src/core/manager.c [hunk 1]
```diff
@@ -4836,22 +4836,15 @@ static int short_uid_range(const char *path) {
...
          * configuration phase. */
 
         assert(m);
 
-        const char* stage[12] = {};
-        size_t n = 0;
-
-        _cleanup_free_ char *usrbin = NULL;
-        if (readlink_malloc("/bin", &usrbin) < 0 || !PATH_IN_SET(usrbin, "usr/bin", "/usr/bin"))
-                stage[n++] = "unmerged-usr";
```
Comment (poettering) [inline src/core/manager.c:4850]: hmm, so we require /usr/ to be pre-mounted now during the initrd→host transition if split off. 

But that's a different thing from requiring that /bin being symlinked to /usr/bin

i tink this check should stay

File: src/core/manager.c [hunk 2]
```diff
@@ -4836,22 +4836,15 @@ static int short_uid_range(const char *path) {
...
-
-        if (access("/proc/cgroups", F_OK) < 0)
-                stage[n++] = "cgroups-missing";
```
Comment (poettering) [inline src/core/manager.c:4853]: i guess this one can be dropped indeed. checking the code we refuse to boot without cgroups these days. looks good hence.

Comment (poettering) [reply src/core/manager.c:4853]: the string is mentioned in man/org.freedesktop.systemd1.xml and catalog/systemd.catalog.in which need to be updated too

File: src/core/manager.c [hunk 3]
```diff
@@ -4854,9 +4854,7 @@ char* manager_taint_string(const Manager *m) {
         if (os_release_support_ended(NULL, /* quiet= */ true, NULL) > 0)
                 stage[n++] = "support-ended";
 
-        _cleanup_free_ char *destination = NULL;
-        if (readlink_malloc("/var/run", &destination) < 0 ||
-            !PATH_IN_SET(destination, "../run", "/run"))
+        if (inode_same("/run", "/var/run", AT_NO_AUTOMOUNT) <= 0)
```
Comment (poettering) [inline src/core/manager.c]: i think for the taint thing there's value in being particularly strict, i.e. only allow two rlevant strings and nothing else. 

i.e. i think a verifier (such as the tain string stuff) should be stricter than an executor (such as the $PATH generation logic). i.e. given that the verifier doesn't immediately cause stuff to fail or change, but just report weirdness, I think there's value in making it super strict i.e. check the exact strings of the symlinks

**Comment by poettering** [comment]
we might want to add a new taint btw, check that /usr/sbin is a symlink to bin, and if not report a taint, as a gentle push for downstreams to merge thier bin+sbin, too.

File: src/core/manager.c [hunk 4]
```diff
@@ -4836,22 +4836,15 @@ static int short_uid_range(const char *path) {
...
          * runtime should be tagged here. For stuff that is known during compilation, emit a warning in the
          * configuration phase. */
 
         assert(m);
 
-        const char* stage[12] = {};
-        size_t n = 0;
-
-        _cleanup_free_ char *usrbin = NULL;
-        if (readlink_malloc("/bin", &usrbin) < 0 || !PATH_IN_SET(usrbin, "usr/bin", "/usr/bin"))
```
Comment (bluca) [inline src/core/manager.c:4849]: I still get people who hack this from time to time, so I'd like to keep this around until Debian 13 is out

File: man/org.freedesktop.systemd1.xml [hunk 5]
```diff
@@ -1665,13 +1665,21 @@ node /org/freedesktop/systemd1 {
           </ulink>.</para>
 
           <xi:include href="version-info.xml" xpointer="v252"/></listitem>
+        </varlistentry>
 
+        <varlistentry>
+          <term><literal>var-run-bad</literal></term>
+
+          <listitem><para><filename>/run/</filename> does not exist or <filename>/var/run</filename> is not a
```
Comment (poettering) [inline man/org.freedesktop.systemd1.xml:1682]: suffix dir paths with / please

File: man/org.freedesktop.systemd1.xml [hunk 6]
```diff
@@ -1665,21 +1665,30 @@ node /org/freedesktop/systemd1 {
           </ulink>.</para>
 
           <xi:include href="version-info.xml" xpointer="v252"/></listitem>
+        </varlistentry>
+
+        <varlistentry>
+          <term><literal>unmerged-bin</literal></term>
+
+          <listitem><para><filename>/usr/sbin</filename> is not a symlink to <filename>/usr/bin</filename>.
```
Comment (poettering) [inline man/org.freedesktop.systemd1.xml]: suffix dir path with / please

**Comment by poettering** [comment]
thanks! lgtm!

File: man/org.freedesktop.systemd1.xml (hunk 5 above)
Comment (YHNdnzj) [reply man/org.freedesktop.systemd1.xml:1682]: So we're referring to the symlink here, not the actual directory. I think / suffix would be somehow misleading?

**Comment by YHNdnzj** [comment]
Updated to suffix all target dirs with `/`. I'm not sure about source as stated above.

File: man/org.freedesktop.systemd1.xml (hunk 5 above)
Comment (yuwata) [reply man/org.freedesktop.systemd1.xml:1682]: I agree with @YHNdnzj here. At least, the sentence is not introduced by this PR. Let's suffix them later if @poettering really wants.

**Pull request merged**